lib/linter.js

```diff
@@ -517,7 +519,7 @@ function getRuleOptions(ruleConfig) {
  * @param {Object} providedParserOptions Options to pass to the parser
  * @param {string} parserName The name of the parser
  * @param {string} filePath The path to the file being parsed.
- * @returns {{success: false, error: Problem}|{success: true,ast: ASTNode, services: Object}}
+ * @returns {{success: false, error: Problem}|{success: true,ast: ASTNode, services: (Object|undefined), scopeManager: (Object|undefined), visitorKeys: (Object|undefined)}}
```

Comment:
I think `services` is always an object here, because if no services are provided, `parse` returns `services: {}`.